Src/Chill/GivenWhenThen.cs

```diff
@@ -183,30 +184,30 @@ protected void WhenLater(Func<Task> whenActionAsync)
         /// <param name="deferredExecution">Should the test be executed immediately or be deferred?</param>
         public void When(Func<Task> whenActionAsync, bool? deferredExecution = null)
         {
-            this.When(() => whenActionAsync.ExecuteInDefaultSynchronizationContext(), deferredExecution);
+            When(() => whenActionAsync.ExecuteInDefaultSynchronizationContext(), deferredExecution);
         }
 
         internal override void TriggerTest(bool expectExceptions)
         {
-            TriggerTest(() => whenAction(), expectExceptions);
+            TriggerTest(() => whenAction.ExecuteInDefaultSynchronizationContext(), expectExceptions);
```

Comment:
We already call ExeuteInDefaultSync... inside TriggerTest in the base class, no need to do it here again